docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -88,142 +158,155 @@ Create the `package.json` file in the root of your project and add the following
 
-<!--SNIPSTART typescript-hello-activity-->
-[hello-world/src/activities.ts](https://github.com/temporalio/samples-typescript/blob/main/hello-world/src/activities.ts)
 ```ts
-export async function greet(name: string): Promise<string> {
-  return `Hello, ${name}!`;
+// The Temporal Workflow.
+export async function getAddressFromIP(name: string): Promise<string> {
+
+  try {
+    const ip = await getIP();
+    try {
+      const location = await getLocationInfo(ip);
+      return `Hello, ${name}. Your IP is ${ip} and your location is ${location}`;
+    } catch (e) {
+      throw new workflow.ApplicationFailure("Failed to get location");
+    }
+  } catch (e) {
+    throw new workflow.ApplicationFailure("Failed to get IP");
+  }
+
 }
 ```
-<!--SNIPEND-->
 
-You've completed the logic for the application; you have a Workflow and an Activity defined. Next, you'll write code to configure and launch a Worker.
+This code uses a `try/catch` block, but because you've configured unlimited retries, there won't be any exceptions caught. However, if you change the Retry Policy's maximum retries, or you specify specific non-retryable exceptions, this code will be in place to handle those errors.
+
+Next you'll create a Worker that executes the Workflow and Activities.
 
 ## Configure and run a Worker
 
-A [Worker](https://docs.temporal.io/concepts/what-is-a-worker) hosts Workflow and Activity functions and executes them one at a time. The Temporal Server tells the Worker to execute a specific function from information it pulls from the [Task Queue](https://docs.temporal.io/concepts/what-is-a-task-queue). After the Worker runs the code, it communicates the results back to the Temporal Server.
+When you start a Temporal Workflow, the Workflow and its Activities get scheduled on the Temporal Service's [Task Queue](https://docs.temporal.io/concepts/what-is-a-task-queue).  A [Worker](https://docs.temporal.io/concepts/what-is-a-worker) hosts Workflow and Activity functions and polls the Task Queue for tasks related to those Workflows and Activities. After the Worker runs the code, it communicates the results back to the Temporal Service where they're stored in the Event History.
 
-Create a file called `worker.ts` and add the following code to define the Worker:
+To define a Worker, you use the Temporal SDK to define a Worker Program.
 
-<!--SNIPSTART typescript-hello-worker-->
-[hello-world/src/worker.ts](https://github.com/temporalio/samples-typescript/blob/main/hello-world/src/worker.ts)
-```ts
+You'll need to specify the name of the Task Queue in your Worker Program, and any time you interact with a Workflow from a client application. You specify the Task Queue name as a case-insensitive string. Since you'll use this string in multiple places, create a constant to guarantee you use it consistently.
```

Comment:
I personally felt the first sentence a little unclear. How about something like:

```suggestion
In your Worker Program, you need to specify the name of the Task Queue, which must match the Task Queue name used whenever you interact with a Workflow from a client application. This ensures that the Worker processes tasks for the correct Workflow. The Task Queue name is a case-insensitive string. To ensure consistency and avoid errors, define the Task Queue name as a constant that can be reused throughout your code.
```

Reply:
love it.